### PR TITLE
Entry Management System: Add tests for Archive and Unarchive

### PR DESCRIPTION
**Pull request by epicfailname**
#### Summary of changes
* Enable `find` command for archives context
* Add unit tests for `ArchiveCommand` and `UnarchiveCommand`
* Add system tests for archive/unarchive

### CODE REVIEW

File: src/test/java/systemtests/EntryBookSystemTest.java [hunk 1]
```diff
@@ -173,19 +173,26 @@ protected void executeCommand(String command) {
...
+        executeCommand(ArchivesCommand.COMMAND_WORD);
+        assertEquals(getModel().getArchivesEntryBook().getEntryList().size(), getModel().getFilteredEntryList().size());
+    }
+
     /**
      * Displays all entries with any parts of their titles matching {@code keyword} (case-insensitive).
      */
     protected void showEntriesWithTitle(String keyword) {
         executeCommand(FindCommand.COMMAND_WORD + " " + PREFIX_TITLE + keyword);
-        assertTrue(getModel().getFilteredEntryList().size() < getModel().getListEntryBook().getEntryList().size());
```
Comment (epicfailname) [inline src/test/java/systemtests/EntryBookSystemTest.java:188]: This line is removed because this is not true all the time, even back in AB4.

Comment (thomastanck) [reply src/test/java/systemtests/EntryBookSystemTest.java:188]: image

File: src/test/java/systemtests/ArchiveCommandSystemTest.java [hunk 2]
```diff
@@ -1,7 +1,310 @@
...
+
+        /* Case: filtered entry list, archive index within bounds of entry book but out of bounds of entry list
+         * -> rejected
+         */
+        showEntriesWithTitle(KEYWORD_MATCHING_MEIER);
+        int invalidIndex = getModel().getListEntryBook().getEntryList().size();
+        command = ArchiveCommand.COMMAND_WORD + " " + invalidIndex;
+        assertCommandFailure(command, MESSAGE_INVALID_ENTRY_DISPLAYED_INDEX);
+
+        /* --------------------- Performing archive operation while a entry card is selected ------------------------ */
```
Comment (thomastanck) [inline src/test/java/systemtests/ArchiveCommandSystemTest.java:88]: Need test for when selected entry is first or last entry.

File: src/test/java/systemtests/ArchiveCommandSystemTest.java [hunk 3]
```diff
@@ -1,7 +1,310 @@
...
+        Index lastEntryIndex = getLastIndex(modelBeforeArchivingLast);
+        assertArchiveCommandSuccess(lastEntryIndex);
+
+        /* Case: archive the middle entry in the list -> archived */
+        Index middleEntryIndex = getMidIndex(getModel());
+        assertArchiveCommandSuccess(middleEntryIndex);
+
+        /* ------------------ Performing archive operation while a filtered list is being shown -------------------- */
+
+        /* Case: filtered entry list, archive index within bounds of entry book and entry list -> archived */
```
Comment (thomastanck) [inline src/test/java/systemtests/ArchiveCommandSystemTest.java:61]: Should test first/mid/last as well for completeness

File: src/test/java/systemtests/ArchiveCommandSystemTest.java [hunk 4]
```diff
@@ -1,7 +1,310 @@
...
+        expectedResultMessage = String.format(MESSAGE_ARCHIVE_ENTRY_SUCCESS, archivedEntry);
+        assertCommandSuccess(command, expectedModel, expectedResultMessage);
+
+        /* ----------------- Show archives, and change the context of the model ------------------------------------- */
+        executeCommand(ArchivesCommand.COMMAND_WORD);
+        assertTrue(getModel().getContext().equals(ModelContext.CONTEXT_ARCHIVES));
+
+        /* ------------------ Performing archive operation while a filtered list is being shown -------------------- */
+
+        /* Case: filtered entry list, archive index within bounds of entry book and entry list -> archived */
```
Comment (thomastanck) [inline src/test/java/systemtests/ArchiveCommandSystemTest.java]: Same as above, first/mid/last

File: src/test/java/systemtests/ArchiveCommandSystemTest.java [hunk 5]
```diff
@@ -1,7 +1,310 @@
...
+        lastEntryIndex = getLastIndex(modelBeforeUnarchivingLast);
+        assertUnarchiveCommandSuccess(lastEntryIndex);
+
+        /* Case: archive the middle entry in the list -> archived */
+        middleEntryIndex = getMidIndex(getModel());
+        assertUnarchiveCommandSuccess(middleEntryIndex);
+
+        /* --------------------- Performing archive operation while a entry card is selected ------------------------ */
+
+        /* Case: archive the selected entry -> entry list panel selects the entry before the archived entry */
```
Comment (thomastanck) [inline src/test/java/systemtests/ArchiveCommandSystemTest.java]: Same as above, should test first/last entry being selected

**Review by thomastanck — APPROVED** [review]
LGTM

**Pull request merged**